src/others/snapshot/lv_snapshot.c

```diff
@@ -111,7 +111,11 @@ lv_result_t lv_snapshot_take_to_buf(lv_obj_t * obj, lv_color_format_t cf, lv_ima
     lv_layer_t layer;
     lv_memzero(&layer, sizeof(layer));
 
-    layer.buf = buf;
+    lv_draw_buf_t draw_buf;
+    lv_memzero(&draw_buf, sizeof(draw_buf));
```

Comment:
Let's change the API `lv_snapshot_take_to_buf ` to lv_snapshot_take_to_buf(obj, draw_buf) in future.
For now you can use `lv_draw_buf_from_image(&draw_buf, dsc);` directly.